spec/lib/jll_property_valuation/api_spec.rb

```diff
@@ -1,951 +1,823 @@
+        "BedroomCount",
+        "StartYear",
+        "StartMonth",
+        "EndYear",
+        "EndMonth",
+        "ListingCount",
+        "ErrorMessage"
+      )
+    end
+  end
+
+  describe '#market_lease_listing_price_median' do
+    let(:body) {
+      {
+        "SuburbName": "",
+        "PostcodeName": "2305",
+        "StateName": "NSW",
+        "StatisticalArea3Id": "",
+        "StatisticalArea4Id": "",
+        "GreaterCapitalCityStatisticalAreaId": "",
+        "PropertyTypeId": "hs",
+        "BedroomCount": 3,
+        "Date": "2022-01-01T00:00:00"
+      }
+    }
+
+    it "returns the correct keys" do
+      expect(subject.market_lease_listing_price_median(body).keys).to include("Status", "Message", "Data")
+    end
+
+    it "returns the correct status" do
+      expect(subject.market_lease_listing_price_median(body)["Status"]).to eq("ok")
+    end
+
+    it "returns the correct message" do
+      expect(subject.market_lease_listing_price_median(body)["Message"]).to eq(nil)
+    end
+
+    it "returns the correct data keys" do
+      expect(subject.market_lease_listing_price_median(body)["Data"].keys).to include(
+        "SuburbName",
+        "PostcodeName",
+        "StatisticalArea3Id",
+        "StatisticalArea4Id",
+        "GreaterCapitalCityStatisticalAreaId",
+        "PropertyTypeId",
+        "BedroomCount",
+        "StartYear",
+        "StartMonth",
+        "EndYear",
+        "EndMonth",
+        "ListingCount",
+        "ListingMedian",
+        "ErrorMessage"
+      )
+    end
+  end
+
+  describe '#market_lease_listing_price_median_trend' do
+    let(:body) {
+      {
+        "SuburbName": "",
+        "PostcodeName": "2305",
+        "StateName": "NSW",
+        "StatisticalArea3Id": "",
+        "StatisticalArea4Id": "",
+        "GreaterCapitalCityStatisticalAreaId": "",
+        "PropertyTypeId": "hs",
+        "BedroomCount": 3,
+        "DateMin": "2015-01-01T00:00:00",
+        "DateMax": "2020-01-01T00:00:00"
+      }
+    }
+
+    it "returns the correct keys" do
+      expect(subject.market_lease_listing_price_median_trend(body).keys).to include("Status", "Message", "Data")
+    end
+
+    it "returns the correct status" do
+      expect(subject.market_lease_listing_price_median_trend(body)["Status"]).to eq("ok")
+    end
+
+    it "returns the correct message" do
+      expect(subject.market_lease_listing_price_median_trend(body)["Message"]).to eq(nil)
+    end
+
+    it "returns the correct data keys" do
+      expect(subject.market_lease_listing_price_median_trend(body)["Data"].keys).to include(
+        "SuburbName",
+        "PostcodeName",
+        "StateName",
+        "StatisticalArea3Id",
+        "StatisticalArea4Id",
+        "GreaterCapitalCityStatisticalAreaId",
+        "PropertyTypeId",
+        "BedroomCount",
+        "DateMin",
+        "DateMax",
+        "LeaseListingIndexItems"
+      )
+    end
+  end
+
+  describe '#market_yield_trend' do
+    let(:body) {
```

Comment:
yeah these are all the same. Try something like

behaves_like_a_jll_request do {body goes here}, [expected data keys go here]

there should be exaples in the shared specs, or I can send you some from conductor (there definitely were in the old specs in EL)